mesonbuild/modules/hotdoc.py

```diff
@@ -0,0 +1,316 @@
+import os
+
+from mesonbuild import mesonlib
+from mesonbuild import mlog, build
+from mesonbuild.coredata import MesonException
+from . import ModuleReturnValue
+from . import ExtensionModule
+from . import get_include_args
+from ..dependencies import Dependency, InternalDependency, ExternalProgram
+
+
+def ensure_list(value):
+    if not isinstance(value, list):
+        return [value]
+    return value
+
+
+MIN_HOTDOC_VERSION = '0.8.100'
+
+
+class HotdocTargetBuilder:
```

Comment:
As I understand it, this builder object is simply used to hold intermediary state while processing the arguments, which is why it's separate from the actual target classes, and why you don't simply have a huge function in the module to prepare the targets. That makes sense to me, but I would suggest following that approach to its end, and simply have something like:

```
builder = Builder()
targets = builder.make_targets()
```

in the `generate_doc()` method of the module, makes sense?

Reply:
Makes sense, I started having a generic Builder that could be reused in other modules, but that experiment didn't work well, let do what you suggest, it will be cleaner :-)